File: app/src/main/java/com/example/myappcontacts/data/providers/database/SQLiteDbProvider.java
package com.example.myappcontacts.data.providers.database;

import com.example.myappcontacts.data.dao.contacts.db.ContactsModel;
import com.example.myappcontacts.data.dao.contactsmap.MapMarkersModel;
import com.example.myappcontacts.data.providers.database.MyStorIOSQLite.IMyStorIOSQLite;

import java.util.List;
import java.util.UUID;

import io.reactivex.Completable;
import io.reactivex.Single;

public class SQLiteDbProvider implements IDataBaseProvider {

    private IMyStorIOSQLite mDataBase;

    public SQLiteDbProvider(IMyStorIOSQLite dataBase) {
        mDataBase = dataBase;
    }

    @Override
    public Single<UUID> addContact() {
        return Single.fromCallable(() -> mDataBase.addContact());
    }

    @Override
    public Completable updateContact(ContactsModel contactsModel) {
        return Completable.fromAction(() -> mDataBase.updateContact(contactsModel));
    }

    @Override
    public Single<ContactsModel> loadContact(UUID contactId) {
        return Single.fromCallable(() -> mDataBase.getContact(contactId));
    }

    @Override
    public Single<List<ContactsModel>> loadContactsList() {
        return Single.fromCallable(() -> mDataBase.getContactsList());
    }

    @Override
    public Completable deleteContact(UUID contactId) {
        return Completable.fromAction(() -> mDataBase.deleteContact(contactId));
    }

    @Override
    public Single<List<MapMarkersModel>> loadMapMarkersList() {
        return Single.fromCallable(() -> mDataBase.getMapMarekersList());
    }
}

File: app/src/main/java/com/example/myappcontacts/data/providers/database/SQLiteDbProvider.kt
package com.example.myappcontacts.data.providers.database

import com.example.myappcontacts.data.dao.contacts.db.ContactsModel
import com.example.myappcontacts.data.dao.contactsmap.db.MapMarkersModel
import com.example.myappcontacts.data.providers.database.MyStorIOSQLite.IMyStorIOSQLite
import io.reactivex.Completable
import io.reactivex.Single
import java.util.*

class SQLiteDbProvider(private val dataBase: IMyStorIOSQLite) : IDataBaseProvider {

    override fun addContact(): Single<UUID> {
        return Single.fromCallable { dataBase.addContact() }
    }

    override fun updateContact(contactsModel: ContactsModel): Completable {
        return Completable.fromAction { dataBase.updateContact(contactsModel) }
    }

    override fun loadContact(contactId: UUID): Single<ContactsModel> {
        return Single.fromCallable { dataBase.getContact(contactId) }
    }

    override fun loadContactsList(): Single<List<ContactsModel>> {
        return Single.fromCallable { dataBase.getContactsList() }
    }

    override fun deleteContact(contactId: UUID): Completable {
        return Completable.fromAction { dataBase.deleteContact(contactId) }
    }

    override fun loadMapMarkersList(): Single<List<MapMarkersModel>> {
        return Single.fromCallable { dataBase.getMapMarkersList() }
    }
}